core/src/main/scala/org/apache/spark/scheduler/TaskLocation.scala

```diff
@@ -62,12 +62,5 @@ private[spark] object TaskLocation {
    * These strings have the form [hostname] or hdfs_cache_[hostname], depending on whether the
    * location is cached.
    */
-  def apply(str: String): TaskLocation = {
-    val hstr = str.stripPrefix(inMemoryLocationTag)
-    if (hstr.equals(str)) {
-      new HostTaskLocation(str)
-    } else {
-      new HostTaskLocation(hstr)
-    }
-  }
+  def apply(str: String): TaskLocation = new HostTaskLocation(str.stripPrefix(inMemoryLocationTag))
```

Comment:
Seems OK, but this seems too minor to bother with the overhead of reviewing and testing. Maybe you can consider changes like this as part of larger logical changes in the future?